src/ViewRendererBundle/objects/helper/TPkgViewRendererLessCompiler.php

```diff
@@ -139,15 +131,18 @@ public function getCompiledCssFilenamePattern()
      */
     protected function getLessFilesFromSnippetsAndTheme($portal)
     {
-        $resourceCollector = new TPkgViewRendererSnippetResourceCollector();
-        $resources = $resourceCollector->getLessResources($portal, CMS_SNIPPET_PATH);
-
         $theme = null;
-        if (is_object($portal)) {
+        if (null !== $portal) {
             $theme = $portal->GetFieldPkgCmsTheme();
         }
 
-        if (!$theme || empty($theme->fieldLessFile)) {
+        $resourceCollector = new TPkgViewRendererSnippetResourceCollector();
+        $this->themeService->setOverrideTheme($theme);
```

Comment:
why are we replacing the theme here? I think the only place we need to do that is in the console command that compiles less for all portals and the backend.

Reply:
ok